File: CS/ChartsDemo.Wpf/ViewModels/NestedDonutViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ChartsDemo {
    public class NestedDonutViewModel {
        const string AgeDataMember = "Age";
        const string GenderDataMember = "Sex";

        public virtual string ArgumentDataMember { get; set; }
        public virtual string SeriesDataMember { get; set; }
        public virtual string DemoTitle { get; set; }
        public virtual string HintDataMember { get; set; }
        public object DataSource { get; private set; }

        public NestedDonutViewModel() {
            DataSource = AgeStructure.GetPopulationAgeStructure();
            ArgumentDataMember = AgeDataMember;
        }

        protected void OnArgumentDataMemberChanged() {
            HintDataMember = (ArgumentDataMember == AgeDataMember) ? GenderDataMember : AgeDataMember;
            SeriesDataMember = "Country" + HintDataMember + "Key";
            DemoTitle = "Population: " + ArgumentDataMember + " Structure";
        }
    }
}

File: VB/ChartsDemo.Wpf/ViewModels/NestedDonutViewModel.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace ChartsDemo
    Public Class NestedDonutViewModel
        Private Const AgeDataMember As String = "Age"
        Private Const GenderDataMember As String = "Sex"

        Public Overridable Property ArgumentDataMember() As String
        Public Overridable Property SeriesDataMember() As String
        Public Overridable Property DemoTitle() As String
        Public Overridable Property HintDataMember() As String
        Private privateDataSource As Object
        Public Property DataSource() As Object
            Get
                Return privateDataSource
            End Get
            Private Set(ByVal value As Object)
                privateDataSource = value
            End Set
        End Property

        Public Sub New()
            DataSource = AgeStructure.GetPopulationAgeStructure()
            ArgumentDataMember = AgeDataMember
        End Sub

        Protected Sub OnArgumentDataMemberChanged()
            HintDataMember = If(ArgumentDataMember = AgeDataMember, GenderDataMember, AgeDataMember)
            SeriesDataMember = "Country" & HintDataMember & "Key"
            DemoTitle = "Population: " & ArgumentDataMember & " Structure"
        End Sub
    End Class
End Namespace
